lib/compilers/java.js

```diff
@@ -143,7 +144,7 @@ export class JavaCompiler extends BaseCompiler {
                         return null;
                     }
 
-                    if (objResult.stdout.includes('void main(java.lang.String[])')) {
+                    if (objResult.stdout.match(this.mainRegex) != null) {
```

Comment:
Tiny nit: could we use RegExp.test instead? Communicates intent better than match :)

Reply:
Of course!